### PR TITLE
Refactor conf file

### PR DESCRIPTION
**Pull request by ngaruko**
# Description

Trying to remove all non-protractor functions from conf.js for reuse as we switch to new framework.

medic/cht-core#[number]

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/conf.js [hunk 1]
```diff
@@ -40,7 +32,7 @@ const baseConfig = {
       // eg: browser.actions().sendKeys(protractor.Key.TAB).perform()
       // https://github.com/angular/protractor/issues/5261
       w3c: false,
-      args: ['--window-size=1024,768', '--headless', '--disable-gpu'],
+      args: ['--window-size=1024,768','--headless','--disable-gpu'],
```
Comment (dianabarsan) [inline tests/conf.js:35]: I think the spaces you removed here made the text more readable.

File: tests/conf.js [hunk 2]
```diff
@@ -74,117 +61,15 @@ const baseConfig = {
...
-        .get('browser')
-        .then(logs => {
-          const currentSpec = jasmine.currentSpec.fullName;
-          browserLogStream.write(`\n~~~~~~~~~~~ ${currentSpec} ~~~~~~~~~~~~~~~~~~~~~\n\n`);
-          logs
-            .map(log => `[${log.level.name_}] ${log.message}\n`)
-            .forEach(log => browserLogStream.write(log));
-          browserLogStream.write('\n~~~~~~~~~~~~~~~~~~~~~\n\n');
-        });
+      utils.saveBrowserLogs();
```
Comment (dianabarsan) [inline tests/conf.js:67]: I think we need to add a return statement here.

File: tests/utils.js [hunk 3]
```diff
@@ -845,4 +890,74 @@ module.exports = {
...
+      await rpn.post('http://localhost:31337/all/start');
+    }
+
+    await listenForApi();
+    const config = await browser.getProcessedConfig();
+    if (config.suite && config.suite === 'web'){
+      await runAndLog('Settings setup', setupSettings);
+    }
+    await runAndLog('User contact doc setup', setUserContactDoc);
+  },
```
Comment (dianabarsan) [inline tests/utils.js:914]: I think we shouldn't inline these large function directly in the export. It makes browsing the file a lot harder. Please make these separate `consts` and export the variables. 
This applies to prepServices, login (which we should also rename to something more specific like "browserLogin"),  setupUser and saveBrowserLogs

File: tests/utils.js [hunk 4]
```diff
@@ -376,6 +379,48 @@ const waitForSettingsUpdateLogs = (type) => {
...
+  try {
+    await request({ path: '/api/info' });
+  } catch(err) {
+    console.log('API check failed, trying again in 1 second');
+    console.log(err.message);
+    await apiRetry();
+  }
+};
+
+const runAndLog = (msg, func) => {
```
Comment (dianabarsan) [inline tests/utils.js:401]: I think we should rename this to signal that these are API startup messages.

**Review by dianabarsan — CHANGES_REQUESTED** [review]
Nice work! I just have some suggestions about readability and clarity.

**Review by dianabarsan — APPROVED** [review]
Great work! Aside from some missing/extra blanks around signs, before parentheses etc (which improve readability), this is perfect.

**Comment by newtewt** [comment]
I pushed a commit that does the clean up around spacing. I want to consume this in the webdriver migration branch and will merge once the build passes.

**Pull request merged**